### PR TITLE
Allow log messages over 8192 bytes

### LINKED ISSUE #1456: functions:shell not emulating functions
node: 8.13.0
firebase-tools:7.0.2
firebase-admin: 8.2.0

when I enter firebase functions:shell following output is coming 

functions: Emulator started at http://localhost:5001
after that its struck 
I have set the GOOGLE_APPLICATION_CREDENTIALS

### PR DESCRIPTION
**Pull request by samtstern**
### Description

Fixes #1456 
	 
### Scenarios Tested

When the size of a message to stdout is over 8192 bytes you get multiple "data" callbacks or it gets truncated, which messes up our communication with the runtime.

With this change, we chunk messages proactively.

### Sample Commands

N/A

### CODE REVIEW

File: src/emulator/functionsEmulator.ts [hunk 1]
```diff
@@ -775,47 +776,39 @@ export function InvokeRuntime(
...
-      // "--no-warnings",
-      path.join(__dirname, "functionsEmulatorRuntime"),
-      JSON.stringify(frb),
-      opts.serializedTriggers || "",
-    ],
-    { env: { node: nodeBinary, ...opts.env, ...process.env }, cwd: frb.cwd }
-  );
+
+  const args = [
+    // "--no-warnings",
```
Comment (abeisgoat) [inline src/emulator/functionsEmulator.ts]: Probably don't need this

File: src/emulator/functionsEmulator.ts [hunk 2]
```diff
@@ -828,7 +821,9 @@ export function InvokeRuntime(
 
   return {
     exit: new Promise<number>((resolve) => {
-      runtime.on("exit", resolve);
```
Comment (abeisgoat) [inline src/emulator/functionsEmulator.ts]: This change seems like a no-op?

File: src/emulator/types.ts [hunk 3]
```diff
@@ -133,7 +152,50 @@ export class EmulatorLog {
   }
 
   log(): void {
```
Comment (abeisgoat) [inline src/emulator/types.ts]: I'd love to see a comment here or somewhere that explains (to our best knowledge) why we need what appears to be an overly complex manual buffering system.

File: src/emulator/functionsEmulator.ts (hunk 2 above)
Comment (samtstern) [reply src/emulator/functionsEmulator.ts]: Yeah you're right I was in there debugging

File: src/emulator/types.ts (hunk 3 above)
Comment (ryanpbrewster) [reply src/emulator/types.ts]: Can we modify the stdout buffer size: https://stackoverflow.com/questions/23429499/stdout-buffer-issue-using-node-child-process

Comment (samtstern) [reply src/emulator/types.ts]: It says `Default: 1024 * 1024.` which is definitely not what I was observing.  I think that's the max total buffered data that has not been flushed and I was running into the 8192 per-write limit

**Pull request merged**